internal/service/cloudprovideraccess/resource_cloud_provider_access_authorization.go

```diff
@@ -278,14 +289,12 @@ func authorizeRole(ctx context.Context, client *admin.APIClient, d *schema.Resou
 		req.SetAtlasAzureAppId(targetRole.GetAtlasAzureAppId())
 		req.SetTenantId(targetRole.GetTenantId())
 		req.SetServicePrincipalId(targetRole.GetServicePrincipalId())
-		roleID = targetRole.GetId()
+		req.SetId(roleID)
 	}
 
-	var role *admin.CloudProviderAccessRole
 	var err error
-
 	for i := 0; i < 3; i++ {
-		role, _, err = client.CloudProviderAccessApi.AuthorizeCloudProviderAccessRole(ctx, projectID, roleID, req).Execute()
+		_, _, err = client.CloudProviderAccessApi.AuthorizeCloudProviderAccessRoleWithParams(ctx, &params).Execute()
```

Comment:
curious about this change, worried we might have some unintentional change here in how role id is being set with the new code.

Reply:
re-reading on L313 to keep the same interfacewhen calling `roleToSchemaAuthorization(role)` on L318

Reply:
Update: From this [thread](https://mongodb.slack.com/archives/C01T8F3BJFJ/p1724859760646959?thread_ts=1724285158.455329&cid=C01T8F3BJFJ) I believe we will get the response fixed, so likely these changes will not be needed.

Reply:
after rebasing from the merged master SDK this is no longer relevant